server/api/api/utils.py

```diff
@@ -1121,6 +1122,16 @@ def get_or_create_project_deletion_background_task(
         # therefore doesn't wait for the project deletion to complete.
         wait_for_project_deletion = True
 
+        # Following the last comment, the model monitoring access key should be retrieved before the project deletion.
+        # This key will be used to delete the model monitoring resources associated with the project.
+        model_monitoring_access_key = (
+            server.api.api.endpoints.nuclio.process_model_monitoring_secret(
+                db_session,
+                project.metadata.name,
+                mlrun.common.schemas.model_monitoring.ProjectSecretKeys.ACCESS_KEY,
+            )
+        )
+
```

Comment:
IIUC, this should be outside the if and should always happen (just one less indentation)

Reply:
but this access key is only used for v3io access, so I'm not sure if it can break the delete API for CE users in this case. I'll test it with CE before making this change.

Reply:
I understand, So yes it needs to be in the if, but not in the else:
```
if (
    server.api.utils.helpers.is_request_from_leader(auth_info.projects_role)
    or mlrun.mlconf.httpdb.projects.leader == "mlrun"
):
    # this means the request either came from zebo (or we are CE) and we are actually about to delete the project.
    # you need the secret in this case and to pass it on to the job. You might need to add another `if igz_version` here
elif igz_version and igz_version < semver.VersionInfo.parse("3.5.5"):
    # In this case the request came client and we are not CE, so we are going to send a request to igz and no deletion is happening here.